core/sail/memory/src/test/java/org/eclipse/rdf4j/sail/memory/QueryPlanRetrievalTest.java

```diff
@@ -873,4 +873,38 @@ public void testDotTimed() {
 
 	}
 
+	@Test
+	public void testArbitraryLengthPath() {
```

Comment:
I would have been ok with you deleting this test, but it's nice of you to turn it into a functional test.

Reply:
It's still a useful test to have in place, making sure that the expected explanation is actually produced.